conda_build/metadata.py

```diff
@@ -907,6 +953,59 @@ def toposort(output_metadata_map):
     return result
 
 
+def _toposort_outputs(output_tuples: list[OutputTuple]) -> list[OutputTuple]:
```

Comment:
Is this code mostly copied from elsewhere?

Reply:
Yes, this is a copy of the now deprecated `toposort` function in the same module, we added them as new functions since the function signatures themselves changes, see https://github.com/conda/conda-build/pull/5345#discussion_r1608233731